## Setup

In [ ]:
import pandas as pd
pd.options.display.float_format = '{:.1f}'.format

In [ ]:
ufos_path = 'YOUR PATH HERE'
ufo_sightings = pd.read_csv(ufos_path)

<ipython-input-2-60ea0c9e0214>:2: DtypeWarning: Columns (5,9) have mixed types. Specify dtype option on import or set low_memory=False.
  ufo_sightings = pd.read_csv(ufos_path)


## Convert the "latitude" and "duration (seconds)" columns to numeric:

In [ ]:
ufo_sightings['duration (seconds)'] = pd.to_numeric(ufo_sightings['duration (seconds)'],errors='coerce')
ufo_sightings['latitude'] = pd.to_numeric(ufo_sightings['latitude'],errors='coerce')

## Apply the `describe` method to the DataFrame:

In [ ]:
ufo_sightings.describe(include='all')

datetime     city  state country  shape  duration (seconds)  \
count            80332    80332  74535   70662  78400             80329.0   
unique           69586    19900     67       5     29                 NaN   
top     7/4/2010 22:00  seattle     ca      us  light                 NaN   
freq                36      525   9655   65114  16565                 NaN   
mean               NaN      NaN    NaN     NaN    NaN              9017.2   
std                NaN      NaN    NaN     NaN    NaN            620228.4   
min                NaN      NaN    NaN     NaN    NaN                 0.0   
25%                NaN      NaN    NaN     NaN    NaN                30.0   
50%                NaN      NaN    NaN     NaN    NaN               180.0   
75%                NaN      NaN    NaN     NaN    NaN               600.0   
max                NaN      NaN    NaN     NaN    NaN          97836000.0   

       duration (hours/min)  comments date posted  latitude  longitude  
count                 80332     80317       80332   80331.0    80332.0  
unique                 8349     79997         317       NaN        NaN  
top               5 minutes  Fireball  12/12/2009       NaN        NaN  
freq                   4716        11        1510       NaN        NaN  
mean                    NaN       NaN         NaN      38.1      -86.8  
std                     NaN       NaN         NaN      10.5       39.7  
min                     NaN       NaN         NaN     -82.9     -176.7  
25%                     NaN       NaN         NaN      34.1     -112.1  
50%                     NaN       NaN         NaN      39.4      -87.9  
75%                     NaN       NaN         NaN      42.8      -78.8  
max                     NaN       NaN         NaN      72.7      178.4

## What state had the most sightings?

**California, abbreviated as "ca"**

## How many sightings occurred in that state?

**9,655**

## How many distinct countries are represented in the dataset?

**5**

## Apply the `describe` method to the house sales dataset - including **all** fields, regardless of data type:

In [ ]:
house_path = 'YOUR PATH HERE'
house_sales = pd.read_csv(house_path)

In [ ]:
house_sales.describe(include='all')

id             date     price  bedrooms  bathrooms  \
count       21613.0            21613   21613.0   21613.0    21613.0   
unique          NaN              372       NaN       NaN        NaN   
top             NaN  20140623T000000       NaN       NaN        NaN   
freq            NaN              142       NaN       NaN        NaN   
mean   4580301520.9              NaN  540088.1       3.4        2.1   
std    2876565571.3              NaN  367127.2       0.9        0.8   
min       1000102.0              NaN   75000.0       0.0        0.0   
25%    2123049194.0              NaN  321950.0       3.0        1.8   
50%    3904930410.0              NaN  450000.0       3.0        2.2   
75%    7308900445.0              NaN  645000.0       4.0        2.5   
max    9900000190.0              NaN 7700000.0      33.0        8.0   

        sqft_living  sqft_lot  floors  waterfront    view  ...   grade  \
count       21613.0   21613.0 21613.0     21613.0 21613.0  ... 21613.0   
unique          NaN       NaN     NaN         NaN     NaN  ...     NaN   
top             NaN       NaN     NaN         NaN     NaN  ...     NaN   
freq            NaN       NaN     NaN         NaN     NaN  ...     NaN   
mean         2079.9   15107.0     1.5         0.0     0.2  ...     7.7   
std           918.4   41420.5     0.5         0.1     0.8  ...     1.2   
min           290.0     520.0     1.0         0.0     0.0  ...     1.0   
25%          1427.0    5040.0     1.0         0.0     0.0  ...     7.0   
50%          1910.0    7618.0     1.5         0.0     0.0  ...     7.0   
75%          2550.0   10688.0     2.0         0.0     0.0  ...     8.0   
max         13540.0 1651359.0     3.5         1.0     4.0  ...    13.0   

        sqft_above  sqft_basement  yr_built  yr_renovated  zipcode     lat  \
count      21613.0        21613.0   21613.0       21613.0  21613.0 21613.0   
unique         NaN            NaN       NaN           NaN      NaN     NaN   
top            NaN            NaN       NaN           NaN      NaN     NaN   
freq           NaN            NaN       NaN           NaN      NaN     NaN   
mean        1788.4          291.5    1971.0          84.4  98077.9    47.6   
std          828.1          442.6      29.4         401.7     53.5     0.1   
min          290.0            0.0    1900.0           0.0  98001.0    47.2   
25%         1190.0            0.0    1951.0           0.0  98033.0    47.5   
50%         1560.0            0.0    1975.0           0.0  98065.0    47.6   
75%         2210.0          560.0    1997.0           0.0  98118.0    47.7   
max         9410.0         4820.0    2015.0        2015.0  98199.0    47.8   

          long  sqft_living15  sqft_lot15  
count  21613.0        21613.0     21613.0  
unique     NaN            NaN         NaN  
top        NaN            NaN         NaN  
freq       NaN            NaN         NaN  
mean    -122.2         1986.6     12768.5  
std        0.1          685.4     27304.2  
min     -122.5          399.0       651.0  
25%     -122.3         1490.0      5100.0  
50%     -122.2         1840.0      7620.0  
75%     -122.1         2360.0     10083.0  
max     -121.3         6210.0    871200.0  

[11 rows x 21 columns]

## How many fields are you able to view statistics such as `top`, `freq`, and `unique` for?

**Only one: "date". This is because all the other fields have a numeric data type, for which the `describe` method won't generate `top`, `freq`, and `unique`.**